modules/yardstick/src/main/java/org/apache/ignite/yardstick/cache/load/IgniteCacheRandomOperationBenchmark.java

```diff
@@ -713,229 +846,283 @@ private void doSqlQuery(IgniteCache cache) throws Exception {
     }
 
     /**
-     * Closure for scan query executing.
+     * @param cache Ignite cache.
+     * @param map Parameters map.
+     * @throws Exception If failed.
      */
-    private static class ScanQueryBroadcastClosure implements IgniteRunnable {
-        /**
-         * Ignite node.
-         */
-        @IgniteInstanceResource
-        private Ignite node;
+    private void doContinuousQuery(IgniteCache cache, Map<Object, Object> map) throws Exception {
+        QueryCursor cursor = (QueryCursor)map.get(cache.getName());
+        if (cursor != null)
+            cursor.close();
+        ContinuousQuery qry = new ContinuousQuery();
+        qry.setLocalListener(new ContinuousQueryUpdater());
+        qry.setRemoteFilterFactory(FactoryBuilder.factoryOf(new ContinuousQueryFilter(nextRandom(50, 100))));
+        map.put(cache.getName(), cache.query(qry));
+    }
 
-        /**
-         * Information about partition.
-         */
-        private Map<UUID, List<Integer>> cachePart;
+/**
+ * Continuous query updater class.
+ */
+private static class ContinuousQueryUpdater implements CacheEntryUpdatedListener, Serializable {
 
-        /**
-         * Name of Ignite cache.
-         */
-        private String cacheName;
+    /** {@inheritDoc} */
+    @Override public void onUpdated(Iterable iterable) throws CacheEntryListenerException {
+        for (Object o : iterable)
+            ;
+    }
+}
 
-        /**
-         * @param cacheName Name of Ignite cache.
-         * @param cachePart Partition by node for Ignite cache.
-         */
-        public ScanQueryBroadcastClosure(String cacheName, Map<UUID, List<Integer>> cachePart) {
-            this.cachePart = cachePart;
-            this.cacheName = cacheName;
-        }
+/**
+ * Continuous query filter class.
+ */
+private static class ContinuousQueryFilter implements CacheEntryEventSerializableFilter, Serializable {
 
-        /** {@inheritDoc} */
-        @Override public void run() {
-            IgniteCache cache = node.cache(cacheName);
+    /**
+     * Value.
+     */
+    private int val;
 
-            // Getting a list of the partitions owned by this node.
-            List<Integer> myPartitions = cachePart.get(node.cluster().localNode().id());
+    /**
+     * @param val Value.
+     */
+    ContinuousQueryFilter(int val) {
+        this.val = val;
+    }
 
-            for (Integer part : myPartitions) {
-                if (ThreadLocalRandom.current().nextBoolean())
-                    continue;
+    /** {@inheritDoc} */
+    @Override public boolean evaluate(CacheEntryEvent evt) throws CacheEntryListenerException {
+        return evt.getOldValue() != null && evt.getValue() != null
```

Comment:
I would better create a simpler approach when you have a boolean flag "accept" which value is inverted on each operation. If the flag is "true" then "evaluate" method returns "true" and "false" otherwise.